input-output-hk/daedalus
# [DDW-589] Refactor consolidation status screen epoch calculation

This PR refactors consolidation status screen epoch calculation to fetching the `slotId` from the API.

## Todo:

- [x] Retrieve the `slotId` from the API
- [x] Use [cardanoexplorer](https://cardanoexplorer.com/api/blocks/pages) as a fallback
  - [x] Implement `cardanoexplorer` fallback for all evironments
- [x] Change the current calculation in the Block Consolidation screen
- [x] Remove local `startTime` variable
- [x] Implement acceptance test for fallback epoch calculation

## Testing

[Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1553156138215000)

We are now fetching the `currentEpoch` number via API. Sometimes this information won't be available for a few milliseconds, when the page should be in its "loading" state (see first screenshot): 

* The numbers in parathesis in the explanation text should be hidden.
* The "x **of y** epochs consolidated" should be hidden.
* The progress bar should not display any text and should have a background animation, showing it's in "loading" mode.

### Testing the "loading state" on a normal situation (API works correctly)

1. When the app is loaded, open the Block Consolidation page;
2. Reload the app (Ctrl+R);
3. The page should be displayed in "loading state" for a few milliseconds;
4. The page should display all the information once the `currentEpoch` is fetched from the API.

### Testing the "loading state" when the API fails

Sometimes the API might fail to retrieve the `currentEpoch`, so as a fallback we fetch just once this information from the [cardanoexplorer api](https://cardanoexplorer.com/api/blocks/pages).

To test it:
1. Run daeadalus on [`staging` env](https://github.com/input-output-hk/daedalus#connect-to-staging-cluster)
2. Comment out [NetworkStatusStore.js#L415](https://github.com/input-output-hk/daedalus/blob/e8a8ae57dbe6b25300f1e375e9486b924f0f1b25/source/renderer/app/stores/NetworkStatusStore.js#L415): `// this.currentEpoch = slotId.epoch;`
3. Open the Block Consolidation screen;
4. The page should be displayed in "loading state" for a few milliseconds;
5. Upon receiving the `currentEpoch`, the page should be displayed fully (second screenshot);
6. On development, the page would remain without the `currentEpoch` information.

## Screenshots

### Without the `currentEpoch` information:

![image](https://user-images.githubusercontent.com/1504716/54538062-57398e80-4972-11e9-87a1-442263eb8283.png)

### With the `currentEpoch` information:

![bc](https://user-images.githubusercontent.com/1504716/54141406-5f3e7f00-4404-11e9-8e10-a46edb905338.png)

---

## Review Checklist:

### Basics


### Code Quality

### Testing

### After Review:
